controllers/logicalvolume_controller.go

```diff
@@ -194,16 +196,60 @@ func (r *LogicalVolumeReconciler) createLV(ctx context.Context, log logr.Logger,
 			return nil
 		}
 
-		resp, err := r.lvService.CreateLV(ctx, &proto.CreateLVRequest{Name: string(lv.UID), DeviceClass: lv.Spec.DeviceClass, SizeGb: uint64(reqBytes >> 30)})
-		if err != nil {
-			code, message := extractFromError(err)
-			log.Error(err, message)
-			lv.Status.Code = code
-			lv.Status.Message = message
-			return err
+		var volume *proto.LogicalVolume
+
+		// Create a snapshot LV
+		if lv.Spec.Source != "" {
+			// accessType should be either "readonly" or "readwrite".
+			if lv.Spec.AccessType != "ro" && lv.Spec.AccessType != "rw" {
```

Comment:
lv.Spec.AccessType is optional. If it does not exist, it should default to rw.
There needs to be an additional check for lv.Spec.AccessType != nil